ExtLibs/ArduPilot/parampck.cs

```diff
@@ -118,10 +132,10 @@ private static object decode_value(char type_format, byte[] data, int startIndex
         {
             switch (type_format)
             {
-                case 'b': return data[startIndex];
-                case 'h': return BitConverter.ToUInt16(data, startIndex);
-                case 'i': return BitConverter.ToUInt32(data, startIndex);
-                case 'f': return BitConverter.ToSingle(data, startIndex);
+                case 'b': return (sbyte)data[startIndex];
+                case 'h': return (short)BitConverter.ToUInt16(data, startIndex);

```

Comment:
whats the intent here?

Reply:
I was getting -1 warping to int max. This matches the return types from here:
https://github.com/ArduPilot/MissionPlanner/blob/43712f778807fcaa39527d3cfaaccd2ac4c9cbe5/ExtLibs/Mavlink/MAVLinkParam.cs#L48-L55

Reply:
should we be casting? or reading in the correct format?  ie ToUint16 vs ToInt16 ?

Reply:
Your right, it looks like I just have the wrong conversion. Copy and paste fail on my part.